src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -627,6 +637,55 @@ static void list_command(int connFd, SQLiteDBInterface *sqlite, bool *loop_exit_
     }
 }
 
+static void cypher_ast_command(int connFd, bool *loop_exit)
+{
+
+    string msg_1 = "Input Query :";
+    int result_wr = write(connFd, msg_1.c_str(), msg_1.length());
+    if (result_wr < 0) {
+        frontend_logger.error("Error writing to socket");
+        *loop_exit = true;
+        return;
+    }
+    result_wr = write(connFd, "\r\n", 2);
+    if (result_wr < 0) {
+        frontend_logger.error("Error writing to socket");
+        *loop_exit = true;
+        return;
+    }
+
+    // Get user response.
+    char user_res[FRONTEND_DATA_LENGTH + 1];
+    bzero(user_res, FRONTEND_DATA_LENGTH + 1);
+    read(connFd, user_res, FRONTEND_DATA_LENGTH);
+    string user_res_s(user_res);
+
+    antlr4::ANTLRInputStream input(user_res_s);
+    // Create a lexer from the input
+    CypherLexer lexer(&input);
+
+    // Create a token stream from the lexer
+    antlr4::CommonTokenStream tokens(&lexer);
+
+    // Create a parser from the token stream
+    CypherParser parser(&tokens);
+
+    ASTBuilder ast_builder;
+    auto* ast = any_cast<ASTNode*>(ast_builder.visitOC_Cypher(parser.oC_Cypher()));
+    string result = ast->print(1);
+
+    frontend_logger.log(result, "log");
+
+    SemanticAnalyzer semantic_analyzer;
+    if(semantic_analyzer.analyze(ast))
+    {
+        cout<<"successfully analyzed "<<endl;
```

Comment:
We should not use cout statements here. Please use logger instead.

Reply:
sure sir, removed those and put logger instead

Reply:
Ok